File: 2015/kt/src/test/kotlin/org/adventofcode/Day04Test.kt
package org.adventofcode

import org.junit.Assert.*
import org.junit.Test

class Day04Test {
  private val d = Day04("04")

  @Test
  fun day04Part1Example() {
    val match = d.matcher("abcdef", 5)
    assertEquals(true, match(609043))
    val match2 = d.matcher("pqrstuv", 5)
    assertEquals(true, match2(1048970))
  }

  @Test
  fun day04Part1() {
    assertEquals(254575, d.part1(d.getLine()))
  }

  @Test
  fun day04Part2() {
    assertEquals(1038736, d.part2(d.getLine()))
  }

}

File: 2015/js/day04.test.js
import { md5Match, part1, part2 } from './day04'
import read from './read'

describe('2015 Day 04', () => {

  const secret = read('04')[0]

  it('part1 md5Match', () => {
    expect(md5Match('abcdef', 5)(609043)).toEqual(true)
    expect(md5Match('pqrstuv', 5)(1048970)).toEqual(true)
  })

  it('part1', () => {
    expect(part1(secret)).toEqual(254575)
  })

  it('part2', () => {
    expect(part2(secret)).toEqual(1038736)
  })
})
